src/extension.ts

```diff
@@ -103,3 +152,110 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
 // tslint:disable-next-line:no-empty
 export function deactivateInternal(): void {
 }
+
+async function setupLocalProjectFolder(uri: vscode.Uri, filePath: string, token: string, account: AzureAccount): Promise<void> {
+    const requiredInputs: ILocalDevelopmentRequiredInputs = getAllRequiredInputs(uri.query);
+    const resourceId: string | null = requiredInputs.resourceId;
+    const defaultHostName: string | null = requiredInputs.defaultHostName;
+    const devContainerName: string | null = requiredInputs.devContainerName;
+    const language: string | null = requiredInputs.language;
+    const downloadAppContent: string | null = requiredInputs.downloadAppContent;
+
+    if (resourceId && defaultHostName && devContainerName && language && downloadAppContent) {
+        ext.context.globalState.update(LocalDevelopmentGlobalStates.projectLanguage, getProjectLanguageForLanguage(language));
+        const vsCodeFilePathUri: vscode.Uri = vscode.Uri.file(filePath);
+        const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'temp');
+
+        try {
+            const functionAppName: string = getNameFromId(resourceId, true);
+            const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
+
+            vscode.window.showInformationMessage(localize('settingUpFunctionAppLocalProjInfoMessage', 'Setting up project for function app "${0}" with language "${1}".', functionAppName, language));
+
+            if (downloadAppContent === 'true') {
+                // NOTE: We don't want to download app content for compiled languages.
+                await callWithTelemetryAndErrorHandling('azureFunctions.getFunctionAppMasterKeyAndDownloadContent', async (_actionContext: IActionContext) => {
+                    const listKeyResponse: HttpOperationResponse = await requestUtils.getFunctionAppMasterKey(account.sessions[0], resourceId, token);
+                    await requestUtils.downloadFunctionAppContent(defaultHostName, downloadFilePath, listKeyResponse.parsedBody.masterKey);
+                });
+            }
+
+            const projectFilePathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, `${functionAppName}`);
+            const projectFilePath: string = projectFilePathUri.fsPath;
+            const devContainerFolderPathUri: vscode.Uri = vscode.Uri.joinPath(projectFilePathUri, '.devcontainer');
+            ext.context.globalState.update(LocalDevelopmentGlobalStates.projectFilePath, projectFilePathUri.fsPath);
+            await callWithTelemetryAndErrorHandling('azureFunctions.extractContentAndDownloadDevContainerFiles', async (_actionContext: IActionContext) => {
+                // tslint:disable-next-line: no-unsafe-any
+                await extract(downloadFilePath, { dir: projectFilePath });
```

Comment:
what will happen here if the downloadAppContent is false?

Reply:
It will create a folder with your app name and set it up with the devcontainer for the language of your app.

Reply:
so the extract wont have any issue not getting a zip file?

Reply:
That is correct